### PR TITLE
Bug fix Handle bad NetSerialize

### PR DESCRIPTION
**Pull request by joshuahuburn**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
If we failed to `NetSerialize` a struct in Spatial we would crash on a `checkf`, in native they simply print a warning. Matched this behaviour.

#### Release note
Bugfix: - Fixed a crash when failing to NetSerialize a struct in spatial. Now print a warning instead which matches native Unreal behavior.

#### Tests
Ran with bad serialization and doesn't crash.

How can this be verified by QA?: Game does not crash with NaN data in component updates.

#### Primary reviewers
@m-samiec @padhansell

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialFastArrayNetSerialize.cpp [hunk 1]
```diff
@@ -60,7 +60,12 @@ void SpatialFastArrayNetSerializeCB::NetSerializeStruct(UScriptStruct* Struct, F
 		{
 			bHasUnmapped = true;
 		}
-		checkf(bSuccess, TEXT("NetSerialize on %s failed."), *Struct->GetStructCPPName());
+
+		// Check the success of the serialization and print a warning if it failed. This is how native handles failed serialization.
+		if (!bSuccess)
+		{
+			UE_LOG(LogRep, Warning, TEXT("SpatialFastArrayNetSerialize: NetSerialize %s failed."), *Struct->GetFullName());
```
Comment (Vatyx) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialFastArrayNetSerialize.cpp]: Shouldn't be LogRep, should be our category.

File: SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp [hunk 2]
```diff
@@ -152,7 +152,13 @@ void ComponentFactory::AddProperty(Schema_Object* Object, Schema_FieldId FieldId
 			{
 				bHasUnmapped = true;
 			}
-			checkf(bSuccess, TEXT("NetSerialize on %s failed."), *Struct->GetStructCPPName());
+
+			// Check the success of the serialization and print a warning if it failed. This is how native handles failed serialization.
+			if (!bSuccess)
+			{
+				UE_LOG(LogRep, Warning, TEXT("AddProperty: NetSerialize %s failed."), *Struct->GetFullName());
```
Comment (Vatyx) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp]: We have the Object here as well. Lets add the Object name to the log message

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialFastArrayNetSerialize.cpp [hunk 3]
```diff
@@ -5,6 +5,8 @@
 #include "EngineClasses/SpatialNetBitReader.h"
 #include "EngineClasses/SpatialNetBitWriter.h"
 
+DEFINE_LOG_CATEGORY(LogSpatialFastArraySerialize);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialFastArrayNetSerialize.cpp]: I think this is too granular. Can we absorb this in another log group?

File: SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp [hunk 4]
```diff
@@ -16,6 +16,8 @@
 #include "Utils/RepLayoutUtils.h"
 #include "Utils/InterestFactory.h"
 
+DEFINE_LOG_CATEGORY(LogSpatialComponentFactory);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp]: Likewise

**Pull request merged**